docs/abi.rst

```diff
@@ -0,0 +1,560 @@
+:any:`abi.StaticArray[T,N] <abi.StaticArray>`  :code:`T[N]`           Static if :code:`T` is static     A fixed-length array of :code:`T` with :code:`N` elements
+:any:`abi.Address`                             :code:`address`        Static                            A 32-byte Algorand address. This is an alias for :code:`abi.StaticArray[abi.Byte, Literal[32]]`.
+:any:`abi.DynamicArray[T] <abi.DynamicArray>`  :code:`T[]`            Dynamic                           A variable-length array of :code:`T`
+:any:`abi.String`                              :code:`string`         Dynamic                           A variable-length byte array assumed to contain UTF-8 encoded content. This is an alias for :code:`abi.DynamicArray[abi.Byte]`.
+:any:`abi.Tuple`\*                             :code:`(...)`          Static if all elements are static A tuple of multiple types
+============================================== ====================== ================================= =======================================================================================================================================================
+
+.. note::
+    \*A proper implementation of :any:`abi.Tuple` requires a variable amount of generic arguments. Python 3.11 will support this with the introduction of `PEP 646 - Variadic Generics <https://peps.python.org/pep-0646/>`_, but until then it will not be possible to make :code:`abi.Tuple` a generic type. As a workaround, we have introduced the following subclasses of :code:`abi.Tuple` for fixed amounts of generic arguments:
+
+    * :any:`abi.Tuple0`: a tuple of zero values, :code:`()`
+    * :any:`abi.Tuple1[T1] <abi.Tuple1>`: a tuple of one value, :code:`(T1)`
+    * :any:`abi.Tuple2[T1,T2] <abi.Tuple2>`: a tuple of two values, :code:`(T1,T2)`
+    * :any:`abi.Tuple3[T1,T2,T3] <abi.Tuple3>`: a tuple of three values, :code:`(T1,T2,T3)`
+    * :any:`abi.Tuple4[T1,T2,T3,T4] <abi.Tuple4>`: a tuple of four values, :code:`(T1,T2,T3,T4)`
+    * :any:`abi.Tuple5[T1,T2,T3,T4,T5] <abi.Tuple5>`: a tuple of five values, :code:`(T1,T2,T3,T4,T5)`
+
+These ARC-4 types are not yet supported in PyTeal:
+
+* Non-power-of-2 unsigned integers under 64 bits, i.e. :code:`uint24`, :code:`uint48`, :code:`uint56`
+* Unsigned integers larger than 64 bits
+* Fixed point unsigned integers, i.e. :code:`ufixed<N>x<M>`
+
+Usage
+"""""""""""""""""""""
+
+Setting Values
+''''''''''''''''
+
+All basic types have a :code:`set()` method which can be used to assign a value. The arguments for this method differ depending on the ABI type. For convenience, here are links to the docs for each class's method:
+
+* :any:`abi.Uint.set()`, which is used by all :code:`abi.Uint` classes and :code:`abi.Byte`
+* :any:`abi.Bool.set()`
+* :any:`abi.StaticArray.set()`
+* :any:`abi.Address.set()`
+* :any:`abi.DynamicArray.set()`
+* :any:`abi.String.set()`
```

Comment:
I take it this is _not_ the same as the DynamicArray set method? (unlike the correspondence for Byte using the Uint one)